examples/platforms/efr32mg12/alarm.c

```diff
@@ -190,27 +202,33 @@ void efr32AlarmProcess(otInstance *aInstance)
             new_expires_microsec = (now + dt) * US_IN_MS;
             status = RAIL_SetTimer(gRailHandle, new_expires_microsec, RAIL_TIME_ABSOLUTE, RAILCb_TimerExpired);
 
-            if (status == RAIL_STATUS_NO_ERROR)
-            {
-                sIsRunning = true;
-            }
-            else
+            if (status != RAIL_STATUS_NO_ERROR)
             {
-                sIsRunning = false;
-                otLogCritPlat("Alarm set timer, status: %d, expires: %u, dt: %u, now: %u", status, sAlarmT0 + sAlarmDt,
-                              dt, now);
+                new_expires_microsec = dt * US_IN_MS;
+                status = RAIL_SetTimer(gRailHandle, new_expires_microsec, RAIL_TIME_DELAY, RAILCb_TimerExpired);
```

Comment:
The explanation of this change is handling the RAIL timer, when requesting timer values in the past.  Under what conditions does this happen?

If (t0+dt) is in the past, why set the timer for (now+dt)?  Wouldn't firing the timer now be more accurate?  This change would seemingly introduce some drift by not firing at the expected time.  What modules or features rely on this?

Reply:
This happens if Thread sets a current time +dt =0 request. there is a finite possible that the timer will roll over to the next milli second by the time the rail timer is called - so the timer is set from the current offset.  

This situation is only likely to happen if the timer is on the point of rolling of over and would most likely mean that there is a fraction of a microsecond difference. Doing it in this way keeps the program flow and call context consistent.

I think the bigger question is if OpenThread should be calling a timer set with the current time and a delta of zero

Reply:
@jwhui what is the expected behavior when otPlatAlarmMilliStartAt() is called with t0+dt = now?

Reply:
OpenThread currently expects that the alarm will fire if `t0 + dt` is <= now. This also applies when `dt` is `0`.

It is true that OpenThread may call `otPlatAlarmMilliStartAt()` with `dt = 0`. While we could update OpenThread to only make calls with `dt > 0`, there is still no guarantee that `otPlatAlarmMilliStartAt()` will be called with a `t0 + dt` that happens in the future. For example, the platform may delay OpenThread's execution just before the call to `otPlatAlarmMilliStartAt()`.

Thoughts?

Reply:
That makes sense that it will still expect the call, and that is what this PR is changing.  I'm okay with this if that's a known way that OpenThread is using the call (dt=0).  How critical is the timing if there is drift from any given call?

Reply:
I think some reasonable amount of drift from any given call is OK. What we want to avoid is accumulated drift across multiple calls, which OpenThread should manage by maintaining `t0`. Then the drift from a given call reduces to jitter in the timer callback.